pkg/reconciler/domainmapping/reconciler.go

```diff
@@ -110,6 +134,62 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, dm *v1alpha1.DomainMappi
 	return err
 }
 
+func autoTLSEnabled(ctx context.Context, dm *v1alpha1.DomainMapping) bool {
+	if !config.FromContext(ctx).Network.AutoTLS {
+		return false
+	}
+	logger := logging.FromContext(ctx)
+	annotationValue := dm.Annotations[networking.DisableAutoTLSAnnotationKey]
+	disabledByAnnotation, err := strconv.ParseBool(annotationValue)
+	if annotationValue != "" && err != nil {
+		// Validation should've caught an invalid value here.
+		// If we have one anyway, assume not disabled and log a warning.
+		logger.Warnf("Invalid annotation value for %q. Value: %q",
+			networking.DisableAutoTLSAnnotationKey, annotationValue)
+	}
+
+	return !disabledByAnnotation
+}
+
+func certClass(ctx context.Context) string {
+	return config.FromContext(ctx).Network.DefaultCertificateClass
+}
+
+func (r *Reconciler) tls(ctx context.Context, dm *v1alpha1.DomainMapping) ([]netv1alpha1.IngressTLS, []netv1alpha1.HTTP01Challenge, error) {
+	if !autoTLSEnabled(ctx, dm) {
+		dm.Status.MarkTLSNotEnabled(v1.AutoTLSNotEnabledMessage)
+		return nil, nil, nil
+	}
+
+	acmeChallenges := []netv1alpha1.HTTP01Challenge{}
+	desiredCert := resources.MakeCertificate(dm, certClass(ctx))
+	cert, err := networkaccessor.ReconcileCertificate(ctx, dm, desiredCert, r)
+	if err != nil {
+		if kaccessor.IsNotOwned(err) {
+			dm.Status.MarkCertificateNotOwned(desiredCert.Name)
+		} else {
+			dm.Status.MarkCertificateProvisionFailed(desiredCert.Name)
+		}
+		return nil, nil, err
+	}
+
+	dnsNames := sets.NewString(desiredCert.Spec.DNSNames...)
+	if dnsNames.Has(dm.Name) {
+		dm.Status.URL.Scheme = "https"
+	}
```

Comment:
So you create a map and iterate over all the elements just to see if the item is in the list. Well we can check that instead of iterating over the map.

And below, we can probably just inject `desiredCert.Spec.DNSNames`.

Reply:
Done